protocol/grpc/internal/server.go

```diff
@@ -40,25 +44,38 @@ func (s *server) SayHello(ctx context.Context, in *HelloRequest) (*HelloReply, e
 	return &HelloReply{Message: "Hello " + in.GetName()}, nil
 }
 
-// InitGrpcServer creates global gRPC server.
-func InitGrpcServer() {
-	port := ":30000"
+func (s *server) Reference() string {
+	return "GrpcGreeterImpl"
+}
+
+type Server struct {
+	listener net.Listener
+	server   *grpc.Server
+}
 
-	lis, err := net.Listen("tcp", port)
+func NewServer() (*Server, error) {
+	listener, err := net.Listen("tcp", ":30000")
```

Comment:
why can not config?

Reply:
ok, I add a address as config in test grpc server